integration-tests/src/tests/client/sharding_upgrade.rs

```diff
@@ -407,19 +450,34 @@ impl TestShardUpgradeEnv {
             .get_shard_layout_from_prev_block(&head.last_block_hash)
             .unwrap();
         let block = env.clients[0].chain.get_block(&head.last_block_hash).unwrap();
+
         for (shard_id, chunk_header) in block.chunks().iter().enumerate() {
-            if chunk_header.height_included() == block.header().height() {
-                let outgoing_receipts = env.clients[0]
+            if chunk_header.height_included() != block.header().height() {
+                continue;
+            }
+            let shard_id = shard_id as ShardId;
+
+            for (i, me) in env.validators.iter().enumerate() {
```

Comment:
This is a new addition, why are we adding this now in the test?

Reply:
I discovered that when starting with 4 shards from genesis the validators don't track all shards. I tried to adjust the test to handle that correctly. Eventually I had to give up when I hit the issue with state sync. I'm not going to be fixing that now but we do have it planned for this month. Anyway this code is correct and will make the transition to not all validators tracking all shards easier in the future.